compactor2/src/test_util/display.rs

```diff
@@ -0,0 +1,392 @@
+            row_heading_chars,
+        }
+    }
+
+    /// return how many characters of `self.width_chars` would be consumed by `range` ns
+    fn time_range_to_chars(&self, time_range: i64) -> usize {
+        // avoid divide by zero
+        if self.ns_per_char > 0.0 {
+            (time_range as f64 / self.ns_per_char) as usize
+        } else if time_range > 0 {
+            self.width_chars
+        } else {
+            0
+        }
+    }
+
+    fn format_level(&self, level: &CompactionLevel) -> String {
+        format!(
+            "{:width$}",
+            display_level(level),
+            width = self.width_chars + self.row_heading_chars
+        )
+    }
+
+    /// Formats a single parquet file into a string of `width_chars`
+    /// characters, which tries to visually depict the timge range of
+    /// the file using the width. See docs on [`ParquetFileFormatter`]
+    /// for examples.
+    fn format_file(&self, file: &ParquetFile) -> String {
+        // use try_into to force conversion to usize
+        let time_width = (file.max_time - file.min_time).get();
+
+        // special case "zero" width times
+        let field_width = if self.min_time == self.max_time {
+            self.width_chars
+        } else {
+            self.time_range_to_chars(time_width)
+        }
+        // account for starting and ending '|'
+        .saturating_sub(2);
+
+        // Get compact display of the file, like 'L0.1'
+        // add |--- ---| formatting (based on field width)
+        let file_string = format!("|{:-^width$}|", display_file_id(file), width = field_width);
+        let row_heading = display_format(file, self.show_size);
+
+        // special case "zero" width times
+        if self.min_time == self.max_time {
+            return format!(
+                "{row_heading:width1$}{file_string:^width2$}",
+                width1 = self.row_heading_chars,
+                width2 = self.width_chars,
+            );
+        }
+
+        // otherwise, figure out whitespace padding at start and back
+        // based on the relative start time of the file
+        // assume time from 0
+        let prefix_time_range = file.min_time.get().saturating_sub(self.min_time);
+        let prefix_padding = " ".repeat(self.time_range_to_chars(prefix_time_range));
+
+        // pad the rest with whitespace
+        let postfix_padding_len = self
+            .width_chars
+            .saturating_sub(file_string.len())
+            .saturating_sub(prefix_padding.len());
+        let postfix_padding = " ".repeat(postfix_padding_len);
+
+        format!(
+            "{row_heading:width$}{prefix_padding}{file_string}{postfix_padding}",
+            width = self.row_heading_chars
+        )
+    }
+}
+
+fn display_level(compaction_level: &CompactionLevel) -> &'static str {
+    match compaction_level {
+        CompactionLevel::Initial => "L0",
+        CompactionLevel::FileNonOverlapped => "L1",
+        CompactionLevel::Final => "L2",
+    }
```

Comment:
That's not `'static`, but shorter:

```suggestion
    format!("L{}", *compaction_level as i16)
```

Reply:
I think being explicit with the mapping rather than being clever is better here (as casting to i16 means implicitly assumes that the order of the enum is the same as the level here)

Reply:
The enum levels are explicitly assigned, they are not auto-generated:

https://github.com/influxdata/influxdb_iox/blob/a79b4ec8994c0f49cdf28793d884e4ce5b345fef/data_types/src/lib.rs#L47-L58